packages/ui/src/elements/AnimateHeight/usePatchAnimateHeight.ts

```diff
@@ -64,7 +66,7 @@ export const usePatchAnimateHeight = ({
         container.style.height = ''
       }
     }
-  }, [open, duration, containerRef, browserSupportsKeywordAnimation])
+  }, [open, duration, containerRef, children, browserSupportsKeywordAnimation])
```

Comment:
Adding `children` as a dependency here will cause this effect to run unnecessarily often, and ultimately prevent a smooth animation. It should only run when `open` changes (or in this case, when content height also changes). The proper way to do is probably to use a resize observer to track the content height as its children dynamically render in, and use _that_ as the target height.

Reply:
Yeah that looks good! I _was_ concerned about including children in the deps array and ResizeObserver entirely escaped my mind. It's the better solution here, cheers.